src/block.rs

```diff
@@ -552,61 +552,93 @@ impl Block {
 }
 
 #[derive(Copy, Clone, Debug, PartialEq)]
-pub enum Comparator {
+pub enum Operator {
+    /// Can perhaps be replaced by wrapping the operator in an option.
     None,
-    /// Eq is also Assign
+    /// The = operator. Can be used as assignment, or to open a scope.
     Eq,
-    /// The == operator, which means Eq but cannot be used to assign
-    EEq,
-    /// The ?= operator
-    QEq,
-    Lt,
-    Gt,
-    Le,
-    Ge,
-    Ne,
+    /// The ?= operator Can be used to open a scope on the condition that it exists.
+    MaybeEq,
+    /// ==
+    Equals,
+    /// !=
+    NotEquals,
+    /// <
+    SmallerThan,
+    /// >
+    GreaterThan,
+    /// <=
+    AtMost,
+    /// >=
+    AtLeast,
 }
 
-impl Comparator {
+impl Operator {
     pub fn from_str(s: &str) -> Option<Self> {
         if s == "=" {
-            Some(Comparator::Eq)
+            Some(Operator::Eq)
         } else if s == "?=" {
-            Some(Comparator::QEq)
+            Some(Operator::MaybeEq)
         } else if s == "==" {
-            Some(Comparator::EEq)
+            Some(Operator::Equals)
         } else if s == "<" {
-            Some(Comparator::Lt)
+            Some(Operator::SmallerThan)
         } else if s == ">" {
-            Some(Comparator::Gt)
+            Some(Operator::GreaterThan)
         } else if s == "<=" {
-            Some(Comparator::Le)
+            Some(Operator::AtMost)
         } else if s == ">=" {
-            Some(Comparator::Ge)
+            Some(Operator::AtLeast)
         } else if s == "!=" {
-            Some(Comparator::Ne)
+            Some(Operator::NotEquals)
         } else {
             None
         }
     }
-
     pub fn from_token(token: &Token) -> Option<Self> {
         Self::from_str(token.as_str())
     }
+
+    /// Returns true if the operator is a comparison operator, or if it's an `=` operator,
+    /// which serves as an alias for the equality comparator.
+    pub fn is_comparator(self) -> bool {
+        matches!(
+            self,
+            Operator::Eq
+                | Operator::Equals
+                | Operator::NotEquals
+                | Operator::AtLeast
+                | Operator::AtMost
+                | Operator::GreaterThan
+                | Operator::SmallerThan
+        )
+    }
+    /// Substitutes `==` for `=`.
+    /// This is to allow `=` to be used as an alias for equality,
+    /// even though that's technically incorrect.
+    pub fn to_comparator(self) -> Self {
+        if self == Self::Eq {
+            Self::Equals
+        } else if self.is_comparator() {
+            self
+        } else {
+            panic!("Cannot convert {self} to comparator operator. Check is_comparator() first.");
```

Comment:
I don't like that this can panic. Can we do without?

Reply:
The function was meant to be executed after you've already checked the is_comparator() function, so if used that way it can't ever panic. 

There might be a better way of doing it, maybe returning an option or something, but then you'd have to expect() that and you're right back to a potential panic.

Reply:
I still find this logic strange because they're all comparators (except the None one) and you renamed them to Operator for mysterious reasons :) I would just handle all of them at the call site, instead of using these functions to filter out one or two of them.

Reply:
I'll probably get rid of the None one by making Block contain an enum-vec instead of a tuple-vec, but that's a big change that will touch all the code.

Reply:
If you really want to change it back to Operator we can do that. 

My thinking was thus: they are all operators, so that name is definitely correct. Most are used to compare ordered values, but the assignment operator, None and the question mark operators are not. Therefore, Comparator is a somewhat misleading name.

I don't like the panic there either, I'll see if something can be done.

Reply:
As I stated in a different comment: I didn't realise ?= was a comparison operator too. I changed it back. As for the panic: if we merge the two enum values that whole function can be deleted.